src/software/sensor_fusion/filter/ball_filter.h

```diff
@@ -52,25 +45,67 @@ class BallFilter
 
     /**
      * Creates a new Ball Filter
-     *
-     * @param min_buffer_size The minimum size of the buffer the filter will use to filter
-     * the ball. The buffer will shrink to this size as the ball speeds up
-     * @param min_buffer_size The maximum size of the buffer the filter will use to filter
-     * the ball. The buffer will grow to this size as the ball slows down
      */
-    explicit BallFilter(unsigned int min_buffer_size = DEFAULT_MIN_BUFFER_SIZE,
-                        unsigned int max_buffer_size = DEFAULT_MAX_BUFFER_SIZE);
+    explicit BallFilter();
 
     /**
      * Filters the new ball detection data, and returns the updated state of the ball
-     * given the new data *
-     * @param current_ball_state The current state of the Ball
+     * given the new data
+     *
      * @param new_ball_detections A list of new Ball detections
+     * @param filter_area The area within which the ball filter will work. Any detections
+     * outside of this area will be ignored.
      *
      * @return The updated state of the ball given the new data
```

Comment:
to be consistent with the other method comments, include condition for which std::nullopt is returned

Reply:
good catch. done